app/controllers/graders_controller.rb

```diff
@@ -286,4 +305,16 @@ def unassign_graders(grader_membership_ids)
     end
     Grouping.unassign_tas(grader_membership_ids, grouping_ids, @assignment)
   end
+
+  def find_empty_submissions(grouping_ids)
+    grouping_ids.each do |grouping_id|
+      submission = Submission.find_by_grouping_id(grouping_id)
+      if !submission
+        next
+      elsif !SubmissionFile.where(submission_id: submission.id).exists?
+        grouping_ids.delete(grouping_id)
```

Comment:
Never delete from an iterable when you're iterating through it. But also, see above comment.